packages/datagrid/stories/DynamicDataGrid.component.js

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import random from 'lodash/random';
+import { IconsProvider } from '@talend/react-components';
+
+import DataGrid from '../src/components/';
+import serializer from '../src/components/DatasetSerializer';
+import DefaultRenderer from '../src/components/DefaultCellRenderer/DefaultRenderer.component';
+import DefaultIntCellRenderer from '../src/components/DefaultIntCellRenderer';
+import DefaultPinHeaderRenderer from '../src/components/DefaultPinHeaderRenderer';
+import DefaultCellRenderer from '../src/components/DefaultCellRenderer';
+import DefaultHeaderRenderer from '../src/components/DefaultHeaderRenderer';
+import sample from './sample.json';
+
+const ADD_ITEMS_NUMBER = 4;
+const LOADING_ITEM = {
+	loading: true,
+	value: {},
+};
+
+/**
+ * getRowDataInfos - this function return information about how many row of each type are loaded
+ *
+ * @param  {array} rowData array of rowData
+ * @return {object}        state of the rowData
+ */
+export function getRowDataInfos(rowData) {
+	return rowData.reduce(
+		(acc, item) => {
+			if (item.loading === false && Object.keys(item).length === 2) {
+				// eslint-disable-next-line no-param-reassign
+				acc.notLoaded += 1;
```

Comment:
Just a question but why not using a new ref and immutability like you did here https://github.com/Talend/ui/pull/1834/files#diff-409fdd0e3d8a537b6c9e9f95d2ab7ea7R164 and https://github.com/Talend/ui/pull/1834/files#diff-409fdd0e3d8a537b6c9e9f95d2ab7ea7R149

Reply:
i just take the code in catalog to get the same behavor

Reply:
So I think we can make the code in catalog cleaner...